app/scripts/models/reliers/oauth.js

```diff
@@ -242,15 +199,9 @@ define(function (require, exports, module) {
 
       return self._oAuthClient.getClientInfo(clientId)
         .then(function (serviceInfo) {
-          self.set('serviceName', serviceInfo.name);
-
-          self._validateOAuthValues(serviceInfo);
-
-          // server version always takes precedent over the search parameter
-          self.set('redirectUri', serviceInfo.redirect_uri);
-          self.set('termsUri', serviceInfo.terms_uri);
-          self.set('privacyUri', serviceInfo.privacy_uri);
-          self.set('trusted', serviceInfo.trusted);
+          var result = self.validateUsingSchema(
```

Comment:
Leave as is, we validate data using a schema, it makes sense for `data` to come first.